p2p/node.go

```diff
@@ -124,6 +152,18 @@ func New(ctx context.Context, config Config) (*Node, error) {
 	} else if config.RendezvousString == "" {
 		return nil, errors.New("config.RendezvousString is required")
 	}
+	if config.GlobalPubSubMessageLimit == 0 {
```

Comment:
@fabioberger This is similar to my other PR, but in this case the `core` package doesn't need to know about these config options at all and I'd like to keep it that way (they aren't user-configurable). The number of options spread around are codebase is going to continue to grow and I think we should be okay with creating defaults in certain sub-packages. What do you think?